search/constraint.go

```diff
@@ -249,17 +225,21 @@ func (ec *equalConstraint) filter(ctx context.Context, rg parquet.RowGroup, prim
 		}
 		// Page intersects [from, to] but we might be able to discard it with statistics
 		if cidx.NullPage(i) {
+			if ec.matches(parquet.ValueOf("")) {
+				res = append(res, RowRange{pfrom, pcount})
+			}
 			continue
 		}
+		// If we are not matching the empty string ( which would be satisfied by Null too ), we can
+		// use page statistics to skip rows
 		minv, maxv := cidx.MinValue(i), cidx.MaxValue(i)
-
-		if !ec.val.IsNull() && !maxv.IsNull() && ec.comp(ec.val, maxv) > 0 {
+		if !ec.matches(parquet.ValueOf("")) && !maxv.IsNull() && ec.comp(ec.val, maxv) > 0 {
```

Comment:
What's the difference here if we do `ec.val.IsNull()` or `ec.matches(parquet.ValueOf(""))` for equal constraint?

Reply:
we can just set Equal(col, parquet.Value("")) if we get a `col=""` matcher - it matches promql a bit better